### PR TITLE
added agent logging middleware

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#633

### CODE REVIEW

File: Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var scopedData = new Dictionary<string, object>();
+
+            try
+            {
+                var agentContext = await _agentContextService.GetAgent();
```
Comment (mingazhev) [inline Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs:26]: You can use agentcontextonternal service to avoid exceptions

Comment (DFSko) [reply Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs]: Fixed

File: Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            _logger = logger;
+        }
+
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var scopedData = new Dictionary<string, object>();
+            var (isSuccess, _, agentContext, _) = await _agentContextService.GetAgentInfo();
+
+            if (isSuccess)
```
Comment (mingazhev) [inline Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs:26]: I suggest to create scope only if it's needed, when we have agent context

**Review by mingazhev — APPROVED** [review]
With a comment

File: Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs (hunk 2 above)
Comment (DFSko) [reply Api/Infrastructure/Middlewares/AgentRequestLoggingMiddleware.cs:25]: fixed

**Pull request merged**